jablib/src/main/java/org/jabref/logic/formatter/bibtexfields/ConvertMSCCodesFormatter.java

```diff
@@ -0,0 +1,117 @@
+package org.jabref.logic.formatter.bibtexfields;
+
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.jabref.logic.cleanup.Formatter;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.layout.LayoutFormatter;
+import org.jabref.logic.shared.exception.MscCodeLoadingException;
+import org.jabref.logic.util.MscCodeUtils;
+import org.jabref.model.entry.BibEntryPreferences;
+import org.jabref.model.entry.Keyword;
+import org.jabref.model.entry.KeywordList;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import org.eclipse.jgit.annotations.NonNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConvertMSCCodesFormatter extends Formatter implements LayoutFormatter {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ConvertMSCCodesFormatter.class);
+    private static final BiMap<String, String> MSCMAP;
+    private static boolean conversionPossible;
+    private static BibEntryPreferences preferences;
+
+    public ConvertMSCCodesFormatter() { }
+
+    public ConvertMSCCodesFormatter(BibEntryPreferences pref) {
+        preferences = pref;
+    }
+
+    public static void setPreferences(BibEntryPreferences pref) {
+        preferences = pref;
+    }
+
+    static {
+        MSCMAP = initializeMap();
+    }
+
+    private static HashBiMap<String, String> initializeMap() {
+        HashBiMap<String, String> tempMap = HashBiMap.create();
+
+        URL resourceUrl = ConvertMSCCodesFormatter.class.getClassLoader().getResource("msc_codes.json");
+
+        // Check for valid mapping of msc codes
+        if (resourceUrl == null) {
+            LOGGER.error("Resource not found: msc_codes.json");
+            conversionPossible = false;
+            return tempMap;
+        } else {
+            try {
+                tempMap = MscCodeUtils.loadMscCodesFromJson(resourceUrl).get();
+                if (!tempMap.isEmpty()) {
+                    conversionPossible = true;
+                    return tempMap;
+                }
+            } catch (MscCodeLoadingException e) {
+                LOGGER.error("Error loading MSC codes:", e);
+                conversionPossible = false;
+            }
+            return tempMap;
+        }
+    }
+
+    @NonNull
+    @Override
+    public String format(String text) {
+        if (text.isEmpty() || !conversionPossible) {
+            return text;
+        }
+
+        // get preferences
+        Character dlim = preferences.getKeywordSeparator();
+        Character hdlim = Keyword.DEFAULT_HIERARCHICAL_DELIMITER;
+//        LOGGER.info("Delimiter = {}", dlim);
+//        LOGGER.info("Hierarchical Delimiter = {}", hdlim);
```

Comment:
You can leave this in with `DEBUG` and increase the log level on your side during development (and do not commit this change)

See https://devdocs.jabref.org/code-howtos/logging.html for details.